### DISCUSSION TITLE
Could not find `app` directory on next build

### DISCUSSION THREAD

**Discussion by juniorEdris** [vercel/next.js#50901] [Help] [answered]
### Summary

Hey, I'm encountering an issue with **Next.js v-13.4.3**. I'm trying to use build the app, but it's not working as expected. It's returning me the issue - **`Couldn't find any `pages` or `app` directory. Please create one under the project root at findPagesDir`**. It is appearing on Vercel deployment also. Has anyone experienced similar issues?

Can anyone get me a solution for this?

![image](https://github.com/vercel/next.js/assets/70286969/421a7f5e-1dcf-48c5-b2ab-d1a626285da1)

### Example

https://github.com/juniorEdris/next-prisma-mongodb-tailwind-app.git

**Comment by icyJoseph** [comment]
Where did you get this line from:

```json
    "build": "next build prisma generate",
```

The `next build` command can take an argument, which is the directory where to look for the project. In your case, the way you have it, it is looking for a directory called `prisma`.

```console
npx next build --help

      Description
        Compiles the application for production deployment

      Usage
        $ next build <dir>

      <dir> represents the directory of the Next.js application.
      If no directory is provided, the current directory will be used.

      Options
      --profile                 Can be used to enable React Production Profiling
      --no-lint                 Disable linting
      --no-mangling             Disable mangling
      --experimental-app-only   Only build 'app' routes
```

Perhaps you meant:

```json
    "build": "prisma generate && prisma migrate deploy && next build",
```

Source: https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-vercel

**Reply by juniorEdris** [reply]
> Where did you get this line from:
> 
> ```json
>     "build": "next build prisma generate",
> ```
> 
> The `next build` command can take an argument, which is the directory where to look for the project. In your case, the way you have it, it is looking for a directory called `prisma`.
> 
> ```
> npx next build --help
> 
>       Description
>         Compiles the application for production deployment
> 
>       Usage
>         $ next build <dir>
> 
>       <dir> represents the directory of the Next.js application.
>       If no directory is provided, the current directory will be used.
> 
>       Options
>       --profile                 Can be used to enable React Production Profiling
>       --no-lint                 Disable linting
>       --no-mangling             Disable mangling
>       --experimental-app-only   Only build 'app' routes
> ```
> 
> Perhaps you meant:
> 
> ```json
>     "build": "prisma generate && prisma migrate deploy && next build",
> ```
> 
> Source: https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-vercel

It was suggested by vercel build column to run the `prisma generate` command during the build process.

![image](https://github.com/vercel/next.js/assets/70286969/ef493fb8-d0e4-4456-afac-54654c25fb5f)

**Reply by icyJoseph** [reply, marked as answer]
Aha well, now you know :) don't forget to mark as answered if you think it's clear.

**Reply by juniorEdris** [reply]
> Aha well, now you know :) don't forget to mark as answered if you think it's clear.

Its fixed. 

**"prisma generate && next build"** made it done where **prisma migrate deploy** was not needed as the model was generated and deployed in development. but **prisma generate** was needed to activate the connection between DB and Application. I think

**Comment by Reddy-Sud0** [comment]
npm run build

> nextjs@0.1.0 build
> next build

Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry


> Build error occurred
Error: > Couldn't find any pages or app directory. Please create one under the project root
    at findPagesDir (/home/u280643504/domains/inteliversetech.com/public_html/node_modules/next/dist/lib/find-pages-dir.js:54:15)
    at /home/u280643504/domains/inteliversetech.com/public_html/node_modules/next/dist/build/index.js:235:73
    at async Span.traceAsyncFn (/home/u280643504/domains/inteliversetech.com/public_html/node_modules/next/dist/trace/trace.js:105:20)
    at async build (/home/u280643504/domains/inteliversetech.com/public_html/node_modules/next/dist/build/index.js:185:29)
    at async main (/home/u280643504/domains/inteliversetech.com/public_html/node_modules/next/dist/bin/next:156:5)

If See the image where it local is correct format but i am getting  when build and deploy in the hostinger  acccount 

![Screenshot 2025-02-03 201312](https://github.com/user-attachments/assets/2f6e88bb-09f7-4506-9e8f-9cfdb5572b92)
